persist/fs/write.go

```diff
@@ -262,8 +262,11 @@ func (w *writer) writeCheckpointFile() error {
 	if err != nil {
 		return err
 	}
-	defer fd.Close()
-	return w.digestBuf.WriteDigestToFile(fd, w.digestFdWithDigestContents.Digest().Sum32())
+	if err := w.digestBuf.WriteDigestToFile(fd, w.digestFdWithDigestContents.Digest().Sum32()); err != nil {
```

Comment:
nit: Extract the digest calculation to line before to reduce line length and improve overall readability?

Reply:
sure thing

Reply:
done